apps/ui/src/Pages/Signup/SignUp.tsx

```diff
@@ -0,0 +1,252 @@
+import * as React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { ThemeProvider } from '@mui/material/styles';
+import defaultTheme from '../../themes/default';
+import SigningHeader from '../../Components/Signing/SigningHeader';
+import { PersonRounded, PhoneAndroidRounded } from '@mui/icons-material';
+import {
+  RadioGroup,
+  FormControlLabel,
+  Radio,
+  FormLabel,
+  Button,
+  TextField,
+  Link,
+  Grid,
+  Box,
+  Container,
+} from '@mui/material';
+import { Gander, SignUpUser } from '../../types/sign-up-user';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  validateEmail,
+  validateName,
+  validatePassword,
+  validatePhoneNumber,
+} from '../../utils/signing/validation';
+import { PasswordField } from '../../Components/Signing/Fields/PasswordField';
+import { EmailField } from '../../Components/Signing/Fields/EmailField';
+import { isUndefined } from 'lodash';
+
+const SignUp = () => {
+  const navigate = useNavigate();
+
+  const [firstNameError, setFirstNameError] = useState<boolean>(false);
+  const [lastNameError, setLastNameError] = useState<boolean>(false);
+  const [emailError, setEmailError] = useState<boolean>(false);
+  const [phoneNumberError, setPhoneNumberError] = useState<boolean>(false);
+  const [passwordError, setPasswordError] = useState<boolean>(false);
+  const [gander, setGander] = useState<Gander>();
+
+  const [isSubmitClicked, setSubmitClicked] = useState<boolean>(false);
+  // const [user, setUser] = useState<SignUpUser>();
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setSubmitClicked(true);
+    const data = new FormData(event.currentTarget);
+    const newUser: SignUpUser = {
+      firstName: data.get('firstName') as string,
+      LastName: data.get('LastName') as string,
+      email: data.get('email') as string,
+      password: data.get('password') as string,
+      phoneNumber: data.get('phone') as string,
+      gander: (data.get('gander') as Gander) ?? 'Other',
+    };
+    // setUser(newUser);
+    validateUser(newUser);
+    console.log({ newUser });
+  };
+
+  const validateUser = useCallback((user: SignUpUser) => {
+    isFirstNameError(user.firstName);
+    isLastNameError(user.LastName);
+    isEmailError(user.email);
+    isPhoneNumberError(user.phoneNumber);
+    isPasswordError(user.password);
+  }, []);
+
+  const isFirstNameError = (value: string) => {
+    setFirstNameError(validateName(value));
+  };
+  const isLastNameError = (value: string) => {
+    setLastNameError(validateName(value));
+  };
+  const isEmailError = (value: string) => {
+    setEmailError(validateEmail(value));
+  };
+  const isPhoneNumberError = (value: string) => {
+    setPhoneNumberError(validatePhoneNumber(value));
+  };
+  const isPasswordError = (value: string) => {
+    setPasswordError(validatePassword(value));
+  };
```

Comment:
These functions seems not necessary. You have the validate functions, use them instead of have millions of states here.

Reply:
You can create a dictionary for the error messages and have one state for the keys of the errors and use it. What will happen if we add another field - adding a state for each error field seems too much. You can look online for examples - I am sure there is a better way for this.